components/ILIAS/Contact/classes/class.ilContactGUI.php

```diff
@@ -294,131 +300,106 @@ protected function changeContactsView(): void
-                    $this->refinery->kindlyTo()->listOf($this->refinery->kindlyTo()->int()),
-                    $this->refinery->custom()->constraint(fn(array $a) => $a !== [], fn() => 'Empty array.'),
-                ])
-            );
-
-            $userlist = [];
-            foreach ($inv_usr_ids as $inv_usr_id) {
-                $login = ilObjUser::_lookupLogin($inv_usr_id);
-                $userlist[] = $login;
-            }
-
-            if ($userlist !== []) {
-                $url = ilLink::_getStaticLink($inv_room_ref_id, 'chtr');
-                $content[] = $this->ui_factory->messageBox()->success(
-                    $this->lng->txt('chat_users_have_been_invited') . $this->ui_renderer->render(
-                        $this->ui_factory->listing()->unordered($userlist)
-                    )
-                )->withButtons([
-                    $this->ui_factory->button()->standard($this->lng->txt('goto_invitation_chat'), $url)
-                ]);
-            }
-        }
+        $content = $this->chatroomInvitationMessage();
+        $action = $this->contactAction();
+        $chat_allowed = (bool) (new ilSetting('chatroom'))->get('chat_enabled', '0');
+        $mail_allowed = $this->rbacsystem->checkAccess(
+            'internal_mail',
+            ilMailGlobalServices::getMailObjectRefId()
+        );
 
-        $table = new ilBuddySystemRelationsTableGUI($this, 'showContacts');
-        $table->populate();
-        $content[] = $this->ui_factory->legacy()->content($table->getHTML());
+        $content = array_merge($content, (new RelationsTable(
+            $this->ui_factory,
+            $this->lng,
+            $this->ui_service,
+            $this->http,
+        ))->build(array_merge(
+            $chat_allowed ? ['chat' => $action('multi', 'invite_to_chat', 'inviteToChat')] : [],
+            $mail_allowed ? ['mail' => $action('multi', 'send_mail_to', 'mailToUsers')] : [],
+        ), $this->ctrl->getLinkTarget($this, 'showContacts'), $action));
 
         $this->tpl->setContent($this->ui_renderer->render($content));
         $this->tpl->printToStdout();
     }
 
-    private function showContactRequests(): void
+    private function updateState(): void
     {
-        if (!ilBuddySystem::getInstance()->isEnabled()) {
-            $this->error->raiseError($this->lng->txt('msg_no_perm_read'), $this->error->MESSAGE);
-        }
+        $get = $this->http->wrapper()->query()->retrieve(...);
+
+        $user_ids = $get('contact_user_ids', $this->refinery->byTrying([
+            $this->refinery->null(),
+            $this->refinery->kindlyTo()->listOf($this->refinery->byTrying([
+                $this->refinery->kindlyTo()->int(),
+            ])),
+            $this->refinery->custom()->transformation(
+                fn($s) => is_array($s) && join('', $s) === 'ALL_OBJECTS' ?
+                    array_column(RelationsTable::data(), 'user_id') :
+                    throw new Exception('Nope')
```

Comment:
```suggestion
                fn($s): array => is_array($s) && join('', $s) === 'ALL_OBJECTS' ?
                    array_column(RelationsTable::data(), 'user_id') :
                    throw new Exception('Nope')
```

Please add better error text for the exception.

Reply:
The message in the exception will never because `byTrying` has it's own message that will be used. The exception message doesn't matter therefore.